lib/fileset/default.nix

```diff
@@ -1012,4 +1013,49 @@ in
             submodules = true;
           }
         );
+
+  /**
+    The empty fileset. It can be useful as a default value or as starting accumulator for a folding operation.
+
+    # Type
+
+    ```
+    empty :: FileSet
+    ```
+  */
+  empty = _emptyWithoutBase;
+
+  /**
+    Tests whether a given value is a fileset, or can be used in place of a fileset.
+
+    # Inputs
+
+    `value`
+
+    : The value to test
+
+    # Type
+
+    ```
+    isFileset :: Any -> Bool
+    ```
+
+    # Examples
+    :::{.example}
+    ## `lib.fileset.isFileset` usage example
+
+    ```nix
+    isFileset ./.
+    => true
+
+    isFileset (unions [  ])
+    => true
+
+    isFileset 1
+    => false
+    ```
+
+    :::
+  */
+  isFileset = x: (_coerceResult "" x).success;
```

Comment:
A non-allocating version of this might be preferable for performance.

Reply:
Hum, I'm unsure what would be the way to achieve this. I'm thinking of reworking `_coerceResult` to `_coerceGeneric` or something that gets the `ok` and `error` functions as arguments, and instantiate `_coerce` where `ok = id` and `error = throw` and `isFileset` where `ok = _: true` and `error = _: false`. Is that what you had in mind? Am I missing a cleaner solution?

Reply:
Don't think it's worth it tbh, the fileset library is not that performant anyways (it's not a primary goal either)

Reply:
The change I describe above is pretty easy to implement, though. So if it makes it a bit better, I'm happy to oblige.